lib/Parser.js

```diff
@@ -297,6 +297,44 @@ Parser.prototype.initializeEvaluating = function() {
 			}
 			return new BasicEvaluatedExpression().setString(result).setRange(expr.range);
 		});
+		this.plugin("evaluate TemplateLiteral", function(node) {
+			var i, quasis = node.quasis, exprs = [];
+			for (i = 0; i < quasis.length; i++) {
+				exprs.push(new BasicEvaluatedExpression().setString(quasis[i].value.cooked).setRange(quasis[i].range));
+
+				if (i > 0) {
+					var prevExpr = exprs[exprs.length - 2], lastExpr = exprs[exprs.length - 1];
+					var expr = this.evaluateExpression(node.expressions[i-1]);
+					if (!(expr.isString() || expr.isNumber())) continue;
+
+					prevExpr.setString(prevExpr.string + (expr.isString() ? expr.string : expr.number) + lastExpr.string)
+					prevExpr.setRange([prevExpr.range[0], lastExpr.range[1]]);
+					exprs.pop();
+				}
```

Comment:
This was unfortunately implemented by changing the `"should parse template strings in sync requires"` test to `function t() { console.log(t.toString()); var name = "sync"; /* ... */ }` 😇 

Not sure how to write tests for this.